### PR TITLE
feat!: check that org page is publishable (Resolves #595, #829, #1062)

### LINKED ISSUE #595: Organization: check that page is publishable

### LINKED ISSUE #829: Organization page preview: there is no publish button
When previewing a FRO page there is no publish button. You have to select Edit to go back to the pages and use the publish button there.

### LINKED ISSUE #1062: During community org page creation the refugees/immigrants field is required but not validated
**Prerequisites**

- [x] I've [searched](https://github.com/accessibility-exchange/platform/issues) for existing issues

**Describe the bug**

On step 2 of the creating an org page has a question "Does your organization specifically serve and support refugees and/or immigrants?" that is marked as required. However, it isn't validated and can be left empty. If you successfully save with it unselected, it will be marked as "No".

**To Reproduce**
1. Go to create an org page
2. On step 2 fill all but one other required fields
3. Do not select any of the "Does your organization specifically serve and support refugees and/or immigrants?" radio buttons.
4. Click the "Save" button
5. Notice that there is no validation error for the "Does your organization specifically serve and support refugees and/or immigrants?" field.
6. Fix other validation errors while leaving the "Does your organization specifically serve and support refugees and/or immigrants?" unselected.
7. Click the "Save" button
8. Notice that the page saves successfully with the "Does your organization specifically serve and support refugees and/or immigrants?" make as "No".

**Expected behavior**

**Screenshots**

Only the area types validation error is returned

The Does your organization specifically serve and support refugees and/or immigrants? radio buttons were not selected

**Details**
- Device: 
- Operating system: 
- Browser: 
- Link to affected page: 

**Additional context**

### PR DESCRIPTION
**Pull request by jobara**
Ensures that the organization page has all the required model fields filled out before allowing publishing. The publish button will be disabled until it is in a publishable state.

Also addresses some other related issues to fix the publish state in the preview page for individual pages, and adds the publish button to the FRO preview page.

Resolves: #595 
Resolves: #829
Resolves: #1062 

### Prerequisites

If this PR changes PHP code or dependencies:

- [x] I've run `composer format` and fixed any code formatting issues.
- [x] I've run `composer analyze` and addressed any static analysis issues.
- [x] I've run `php artisan test` and ensured that all tests pass.
- [x] I've run `composer localize` to update localization source files and committed any changes.

If this PR changes CSS or JavaScript code or dependencies:

- [x] I've run `npm run lint` and fixed any linting issues.
- [ ] I've run `npm run build` and ensured that CSS and JavaScript assets can be compiled.

### CODE REVIEW

File: app/Models/Organization.php [hunk 1]
```diff
@@ -298,13 +301,57 @@ public function hasAddedDetails(): bool
...
+            return false;
+        }
+
+        if ($this->extra_attributes['has_indigenous_identities'] && ! $this->indigenousIdentities()->count()) {
+            return false;
+        }
+
+        if ($this->extra_attributes['has_age_brackets'] && ! $this->ageBrackets()->count()) {
+            return false;
+        }
```
Comment (greatislander) [inline app/Models/Organization.php]: I think there needs to be additional checks for has_ethnoracial_identities and has_gender_and_sexual_identities— if either of these are true, the related identities need to be attached (the way you've done with indigenous identities and age brackets).

**Review by greatislander — CHANGES_REQUESTED** [review]

File: app/Models/Organization.php (hunk 1 above)
Comment (jobara) [reply app/Models/Organization.php]: I've added new checks and tests to cover these.

File: app/Models/Organization.php [hunk 2]
```diff
@@ -298,13 +301,65 @@ public function hasAddedDetails(): bool
...
+
+        if ($this->extra_attributes['has_ethnoracial_identities'] && ! $this->ethnoracialIdentities()->count()) {
+            return false;
+        }
+
+        if ($this->extra_attributes['has_gender_and_sexual_identities'] && ! $this->genderIdentities()->count()) {
+            return false;
+        }
```
Comment (greatislander) [inline app/Models/Organization.php]:
```suggestion
        if ($this->extra_attributes['has_gender_and_sexual_identities'] && ! $this->genderIdentities()->count() && !$this->constituencies->contains(Constituency::where('name->en', 'Trans people')->first()) && !$this->consitutencies->contains(Constituency::where('name->en', '2SLGBTQIA+ people')->first())) {
            return false;
        }
```

**Review by greatislander — CHANGES_REQUESTED** [review]
Something like this…

File: app/Models/Organization.php (hunk 2 above)
Comment (greatislander) [reply app/Models/Organization.php]: @jobara You'll want to check those names in the `ConstituencySeeder`, I was guessing at them— I think they are correct though

Comment (jobara) [reply app/Models/Organization.php]: @greatislander I've made the change to incorporate the constituencies into the check.

**Review by greatislander — APPROVED** [review]

**Pull request merged**